Sources/BraintreeThreeDSecure/BTThreeDSecureRequest.swift

```diff
@@ -66,38 +42,76 @@ import BraintreeCore
         get { _cardAddChallenge }
         set { _cardAddChallenge = newValue }
     }
-
+    
     // swiftlint:disable identifier_name
     /// Internal property for `cardAddChallenge`. Created to avoid deprecation warnings upon accessing
     /// `cardAddChallenge` directly within our SDK. Use this value internally instead.
     var _cardAddChallenge: BTThreeDSecureCardAddChallenge = .unspecified
-    // swiftlint:enable identifier_name
-
-    /// Optional.  An authentication created using this flag should only be used for vaulting operations (creation of customers' credit cards or payment methods) and not for creating transactions.
-    /// If set to `true`, a card-add challenge will be requested from the issuer.
-    /// If set to `false`, a card-add challenge will not be requested. 
-    /// If the parameter is missing, a card-add challenge will only be requested for $0 amount.
-    public var cardAddChallengeRequested: Bool = false
-
-    /// Optional. UI Customization for 3DS2 challenge views.
-    public var v2UICustomization: BTThreeDSecureV2UICustomization?
-
-    /// Optional. Sets all UI types that the device supports for displaying specific challenge user interfaces in the 3D Secure challenge.
-    ///
-    /// Defaults to `.both`
-    public var uiType: BTThreeDSecureUIType = .both
-
-    /// Optional. List of all the render types that the device supports for displaying specific challenge user interfaces within the 3D Secure challenge.
-    ///
-    /// - Note: When using `BTThreeDSecureUIType.both` or `BTThreeDSecureUIType.html`, all `BTThreeDSecureRenderType` options must be set.
-    /// When using `BTThreeDSecureUIType.native`, all `BTThreeDSecureRenderType` options except `.html` must be set.
-    public var renderTypes: [BTThreeDSecureRenderType]?
-
+    
     /// A delegate for receiving information about the ThreeDSecure payment flow.
     public weak var threeDSecureRequestDelegate: BTThreeDSecureRequestDelegate?
     
-    // MARK: - Internal Properties
+    // MARK: - Initializer
     
-    /// The dfReferenceID for the session. Exposed for testing.
-    var dfReferenceID: String?
+    /// Creates a `BTThreeDSecureRequest`
+    /// - Parameters:
+    ///    - accountType: Optional. The account type selected by the cardholder. Some cards can be processed using either a credit or debit account and cardholders have the option to choose which account to use.
+    ///    - additionalInformation: Optional. The additional information used for verification.
+    ///    - amount: Required. The amount for the transaction.
```

Comment:
If this is required lets put it first and not provide a default. It's also worth considering if we want to continue to use `NSDecimalNumber` or if we can switch to `Double`?

Reply:
ahh yep, so i just verified with the public docs and it looks like `amount` is a `string` and required (same with the `nonce` property) - updated the initializer and other places in the code accordingly